version: '3.8'

services:

  db:
    container_name: db  
    image: postgres:alpine
    env_file:
    - ./core/.env
    environment:
      - POSTGRES_DB=bahoosh
      - POSTGRES_USER=admin
      - POSTGRES_PASSWORD=mhmdbigdeli031@
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    ports:
      - "5432:5432"
    networks:
      - main

  backend:
    container_name: backend
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    restart: on-failure
    volumes:
      - ./core:/app
    env_file:
    - ./core/.env
    ports:
      - "8000:8000"
    networks:
      - main
    depends_on:
      - redis
      - db
  
  redis:
    container_name: redis
    image: redis
    ports:
      - "6379:6379"
    volumes:
      - redis_data:/data
    restart: unless-stopped
    networks:
      - main

  celery_worker:
    container_name: celery_worker
    build: .
    command: celery -A core worker -l INFO --concurrency=3 -Q celery -E
    env_file:
      - ./core/.env
    environment:
      - DJANGO_SETTINGS_MODULE=core.settings
    depends_on:
      - redis
      - backend
      - db
    restart: unless-stopped
    networks:
      - main

volumes:
  redis_data:
  postgres_data:

networks:
  main:
